src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1147,6 +1205,17 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check if App gives permission to AccountManager
+            try
```

Comment:
I'm sorry if i confuse you. I may misunderstand what you want to achieve here. But this piece of code here will not solve our problems, it won't be hit. Your test should be able to capture this.